### PR TITLE
foodmasst - return food metadata for matching files

### PR DESCRIPTION
**Pull request by ka-west**

### CODE REVIEW

**Comment by ka-west** [comment]
@mwang87 @robinschmid

File: search_single_spectrum/tools/search_single_spectrum/foodomics_track.py [hunk 1]
```diff
@@ -46,18 +46,35 @@ def calculate_enrichment(file_occurrent_df, metadata_df):
 
     return pd.DataFrame(output_list)
 
+def metadata_file_matches(file_occurrent_df):
+    # Filter to appropriate dataset
+    file_occurrent_df = file_occurrent_df[file_occurrent_df["dataset_id"] == "MSV000084900"]
+
+    # Get foodomics full metadata
+    gfop_meta = pd.read_csv('https://raw.githubusercontent.com/ka-west/GFOPontology/master/data/foodomics_metadata_foodmasst.txt', sep='\t')
```
Comment (robinschmid) [inline search_single_spectrum/tools/search_single_spectrum/foodomics_track.py:54]: We might want to create the data, metadata, code repositories first and link out to those files. 
This is the new reduced file right? Is it only for foodmasst or can we name it just foodomics_metadata.txt

Comment (ka-west) [reply search_single_spectrum/tools/search_single_spectrum/foodomics_track.py:54]: Oh yes good point about transferring the repos first. It is only for foodmasst, we want to keep the whole original file as well.

Comment (robinschmid) [reply search_single_spectrum/tools/search_single_spectrum/foodomics_track.py:54]: maybe name it .tsv to signal that it is tab-separated

File: search_single_spectrum/tools/search_single_spectrum/foodomics_track.py [hunk 2]
```diff
@@ -46,18 +46,35 @@ def calculate_enrichment(file_occurrent_df, metadata_df):
 
     return pd.DataFrame(output_list)
 
+def metadata_file_matches(file_occurrent_df):
+    # Filter to appropriate dataset
+    file_occurrent_df = file_occurrent_df[file_occurrent_df["dataset_id"] == "MSV000084900"]
```
Comment (robinschmid) [inline search_single_spectrum/tools/search_single_spectrum/foodomics_track.py:51]: I would put this line together with the file_occurrent_df["basefilename"] = .... line where you remove the extension on this dataframe into a separate method.
and call it in the main before you run the two methods:
filtered_matches = filterMatches(..)

1. you apply the same filtering in both methods and its always easier to maintain if you split the filter logic
2. with python you get weird behavior when passing variables as arguments to methods: Its either pass by reference (changes the original variable if altered in the method) or by value (changes in the method are not reflect outside). To be sure - just pass your dataframe in and return the filtered in the end

Comment (robinschmid) [reply search_single_spectrum/tools/search_single_spectrum/foodomics_track.py]: Done

**Pull request closed without merging**